__tests__/app.test.js

```diff
@@ -67,24 +67,72 @@ describe("GET /api/articles/:article_id", () => {
       });
   });
 
-  test("400: respond with an error message if searched with wrong url. endpoint /api/articles/:article_id", () => {
+  test("400: responds with an error message if searched with wrong url. endpoint /api/articles/:article_id", () => {
     return request(app)
       .get("/api/articles/xyz")
       .expect(400)
       .then(({ body }) => {
-        const {article} = body;
         expect(body.msg).toEqual("Bad request. Please check what you're requesting and try again.");
       });
     });
 
-    test("404: respond with JSON object of all /api/articles/:article_id endpoints", () => {
+    test("404: responds with JSON object of all /api/articles/:article_id endpoints", () => {
       return request(app)
         .get("/api/articles/1500")
         .expect(404)
         .then(({ body }) => {
-          const {article} = body;
           expect(body.msg).toEqual("Record Not Found");
         });
     });
 });
 
+describe("GET /api/articles", () => {
+  test("200: responds with JSON object of all articles", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body }) => {
+        const { articles } = body;
+        articles.forEach((article) => {
+          expect(article).toHaveProperty("article_id", expect.any(Number));
+          expect(article).toHaveProperty("title", expect.any(String));
+          expect(article).toHaveProperty("topic", expect.any(String));
+          expect(article).toHaveProperty("author", expect.any(String));
+          expect(article).toHaveProperty("created_at", expect.any(String));
+          expect(article).toHaveProperty("votes", expect.any(Number));
+          expect(article).toHaveProperty("article_img_url", expect.any(String));
+          expect(article).toHaveProperty("comment_count", expect.any(String));
+          expect(articles).toHaveLength(18);
+          if(article.article_id === 1){
+          expect(article).toEqual({
+            article_id: 1,
+            title: 'Living in the shadow of a great man',
+            topic: 'mitch',
+            author: 'butter_bridge',
+            created_at: '2020-07-09T20:11:00.000Z',
+            votes: 100,
+            article_img_url: 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700',
+            comment_count: '11'
+          });
+        }
+      });
+      });
+  });
```

Comment:
We also need to check the articles are sorted by date in descending order. You may want to look into jest sorted to help you test for this - https://www.npmjs.com/package/jest-sorted

Reply:
Added the test.